app/views/layouts/_social_media.html.erb

```diff
@@ -1,10 +1,13 @@
+<% compact_margin = Settings::General.social_media_handles.values.compact_blank.count > 5 %>
 <% Settings::General.social_media_handles.each do |social_media_type, handle| %>
   <% next if handle.blank? %>
+  <%# There isn't a single Mastodon site to build the target URL so the setting should contain the complete URL %>
+  <% target_url = social_media_type == "mastodon" ? handle : "https://#{social_media_type}.com/#{handle}" %>
   <a
-    href="<%= "https://#{social_media_type}.com/#{handle}" %>"
+    href="<%= target_url %>"
     target="_blank"
-    class="c-link c-link--icon-alone c-link--block mx-1"
-    rel="noopener">
+    class="c-link c-link--icon-alone c-link--block <%= compact_margin ? "" : "mx-1" %>"
+    rel="noopener me">
```

Comment:
`compact_margin` here essentially gets rid of the `mx-1` padding so the icons don't overflow (only in the case more than 5 social media links are enabled)